probably/pgcl/syntax.py

```diff
@@ -119,48 +119,53 @@ def check_is_linear_program(program: Program) -> Optional[CheckFail]:
     """
     for instr_ref in walk_instrs(Walk.DOWN, program.instructions):
         for expr in instr_exprs(instr_ref.val):
-            res = check_is_linear_expr(expr)
+            res = check_is_linear_expr(expr, program)
             if isinstance(res, CheckFail):
                 return res
     return None
 
 
-def check_is_linear_expr(expr: Expr) -> Optional[CheckFail]:
+def check_is_linear_expr(expr: Expr, context: Program) -> Optional[CheckFail]:
```

Comment:
To be consistent with the other functions, please use `Program` as the first argument.